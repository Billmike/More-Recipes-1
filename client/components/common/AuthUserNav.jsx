import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * User navigation when unathenticated
 *
 * @param {any} props 
 * @returns {object} React element
 */
const AuthUserNav = (props) => {
  const { handleLogoutUser } = props;
  return (
    <ul id="user-control" className="dropdown-content">
      <li>
        <a href="profile.html" className="black-text">
          Profile
          <i className="material-icons left">person</i>
        </a>
      </li>
      <li>
        <Link to="/user/recipes" className="black-text">
          My Recipes<i className="material-icons left">folder</i>
        </Link>
      </li>
      <li>
        <a
          href="recipes.html?#user-favorites"
          className="black-text"
        >
          My Favorites
          <i className="material-icons left">folder</i>
        </a>
      </li>
      <li className="divider" />
      <li>
        <a
          role="button"
          tabIndex="0"
          className="black-text"
          onClick={handleLogoutUser}
        >
          Logout
          <i className="material-icons left">lock</i>
        </a>
      </li>
    </ul>
  );
};

AuthUserNav.propTypes = {
  handleLogoutUser: PropTypes.func.isRequired,
};

export default AuthUserNav;
