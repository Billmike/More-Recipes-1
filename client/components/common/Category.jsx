import React from 'react';

/**
 * Category options component
 *
 * @returns {object} React element
 */
const Category = () => (
  <ul id="category" className="dropdown-content">
    <li><a href="#!" className="collection-item black-text">Breakfast</a></li>
    <li><a href="#!" className="collection-item black-text">Lunch</a></li>
    <li><a href="#!" className="collection-item black-text">Dinner</a></li>
    <li><a href="#!" className="collection-item black-text">Appetizer</a></li>
    <li><a href="#!" className="collection-item black-text">Main</a></li>
    <li><a href="#!" className="collection-item black-text">Dessert</a></li>
  </ul>
);

export default Category;
