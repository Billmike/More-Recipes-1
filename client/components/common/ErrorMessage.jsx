import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = (props) => {
  const { message } = props;
  return (
    <div className="red-text center-align">
      {message}
    </div>
  );
};

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
};

export default ErrorMessage;
