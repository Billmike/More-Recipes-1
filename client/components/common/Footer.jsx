import React from 'react';
import { Link } from 'react-router-dom';

/**
 * Footer component
 *
 * @returns {object} React element
 */
const Footer = () => (
  <div className="page-footer deep-orange darken-1">
    <div className="container center-align">
      <div className="row">
        <div className="connect-icons col s12 m6 l6">
          <h5 className="white-text">Stay Connected</h5>
          <i className="fa fa-facebook-square fa-2x" />
          <i className="fa fa-twitter-square fa-2x" />
          <i className="fa fa-pinterest-square fa-2x" />
          <i className="fa fa-instagram fa-2x" />
        </div>
        <div className="col s12 m6 l4 offset-l2">
          <h5 className="white-text">Navigate</h5>
          <ul>
            <li>
              <Link to="/" className="grey-text text-lighten-3">Home</Link>
            </li>
            <li>
              <a className="grey-text text-lighten-3" href="#!">About Us</a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div className="footer-copyright">
      <div className="container">
        Â© 2017 All rights reserved
      </div>
    </div>
  </div>
);

export default Footer;
