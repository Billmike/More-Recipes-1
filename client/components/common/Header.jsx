import React from 'react';

import Navbar from './Navbar';
import SideNav from './SideNav';

const Header = props => (
  <div>
    <div className="navbar-fixed">
      <nav className="deep-orange darken-4">
        <Navbar {...props} />
      </nav>
    </div>
    <SideNav />
  </div>
);

export default Header;
