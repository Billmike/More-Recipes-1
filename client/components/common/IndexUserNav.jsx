import React from 'react';
import PropTypes from 'prop-types';

/**
 * User navigation when unathenticated
 *
 * @param {any} props - The properties passed to the component
 * @returns {object} React element
 */
const IndexUserNav = (props) => {
  const { handleOpenSignin, handleOpenSignup } = props;
  return (
    <ul id="user-control" className="dropdown-content">
      <li>
        <a
          role="button"
          tabIndex="0"
          className="black-text"
          onClick={handleOpenSignin}
        >
          Sign in
          <i className="material-icons left">person</i>
        </a>
      </li>
      <li>
        <a
          role="button"
          tabIndex="0"
          className="black-text"
          onClick={handleOpenSignup}
        >
          Register
          <i className="material-icons left">folder</i>
        </a>
      </li>
    </ul>
  );
};

IndexUserNav.propTypes = {
  handleOpenSignup: PropTypes.func.isRequired,
  handleOpenSignin: PropTypes.func.isRequired,
};

export default IndexUserNav;
