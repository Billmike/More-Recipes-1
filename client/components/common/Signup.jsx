import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import isEmpty from 'lodash/isEmpty';

import * as helpers from '../../utils/validate';
import ErrorMessage from '../common/ErrorMessage';
import RenderField from '../common/RenderField';
import Spinner from '../common/Spinner';

const { validate } = helpers;
const propTypes = {
  openSignup: PropTypes.bool.isRequired,
  handleToggleSignupModal: PropTypes.func.isRequired,
  handleToggleModal: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleSubmitSignup: PropTypes.func.isRequired,
  user: PropTypes.shape({
    isLoading: PropTypes.bool.isRequired,
    error: PropTypes.shape({
      message: PropTypes.string
    })
  }).isRequired,
};

/**
 * Signup component
 *
 * @param {any} props The props passed to component
 *
 * @returns {object} React element
 */
export const Signup = (props) => {
  const {
    user: { isLoading, error }, openSignup, handleToggleSignupModal,
    handleToggleModal, handleSubmit, handleSubmitSignup
  } = props;
  const { message } = error;

  const actions = [
    <FlatButton
      label="Cancel"
      secondary
      onClick={handleToggleSignupModal}
    />
  ];

  return (
    <div>
      <Dialog
        title="Sign Up"
        actions={actions}
        modal
        open={openSignup}
        autoScrollBodyContent
      >
        <div className="row">
          <form
            className="col s12"
            onSubmit={handleSubmit(handleSubmitSignup)}
          >
            <div className="row">
              <div className="input-field col s12">
                <Field
                  name="firstName"
                  label="First Name"
                  component={RenderField}
                  type="text"
                />
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <Field
                  name="lastName"
                  label="Last Name"
                  component={RenderField}
                  type="text"
                />
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <Field
                  name="username"
                  label="Username"
                  component={RenderField}
                  type="text"
                />
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <Field
                  name="email"
                  label="Email"
                  component={RenderField}
                  type="email"
                />
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <Field
                  name="password"
                  label="Password"
                  component={RenderField}
                  type="password"
                />
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <Field
                  name="retypePassword"
                  label="Retype-Password"
                  component={RenderField}
                  type="password"
                />
              </div>
            </div>
            {!isEmpty(error) && <ErrorMessage message={message} />}
            <div className="row" />
            <div className="center-align">
              {isLoading && <Spinner size="small" />}
            </div>
            <div className="row center-align">
              <button
                type="submit"
                name="signupbtn"
                className={`btn btn-large waves-effect waves-light
                  indigo accent-2`}
                disabled={isLoading}
              >
                Sign Up
              </button>
            </div>
            <div className="row center-align">
              Already have an account? <a
                role="button"
                tabIndex="0"
                className="cursor-pointer blue-text"
                onClick={handleToggleModal}
              >
                Sign In
              </a>
            </div>
          </form>
        </div>
      </Dialog>
    </div>
  );
};

Signup.propTypes = propTypes;

export default reduxForm({
  form: 'Signup',
  validate,
})(Signup);
