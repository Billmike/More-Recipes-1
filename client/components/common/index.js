import Header from './Header';
import Footer from './Footer';
import RecipeImage from './RecipeImage';

export default {
  Header,
  Footer,
  RecipeImage,
};
