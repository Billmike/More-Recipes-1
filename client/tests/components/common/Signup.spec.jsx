import React from 'react';

import { Signup } from '../../../components/common/Signup';

describe('Signup component', () => {
  const props = {
    openSignup: true,
    handleToggleSignupModal: jest.fn(),
    handleToggleModal: jest.fn(),
    handleSubmit: jest.fn(),
    handleSubmitSignup: jest.fn(),
    user: {
      isLoading: false,
      error: {},
    },
  };
  it('should render component', () => {
    const wrapper = shallow(<Signup {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should show spinner when submitting details', () => {
    const newProps = {
      ...props,
      user: {
        ...props.user,
        isLoading: true,
        error: {},
      },
    };
    const wrapper = shallow(<Signup {...newProps} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Spinner').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
  it('should show error in signup', () => {
    const newProps = {
      ...props,
      user: {
        ...props.user,
        isLoading: false,
        error: {
          message: 'Error in signup',
        },
      },
    };
    const wrapper = shallow(<Signup {...newProps} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('ErrorMessage').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
