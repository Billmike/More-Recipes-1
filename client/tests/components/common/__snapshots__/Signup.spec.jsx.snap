// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component should render component 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[MockFunction]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={true}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    modal={true}
    open={true}
    repositionOnUpdate={true}
    title="Create an account"
  >
    <div
      className="row"
    >
      <form
        className="col s12"
      >
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="First Name"
              name="firstName"
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Last Name"
              name="lastName"
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Username"
              name="username"
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Email"
              name="email"
              type="email"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Password"
              name="password"
              type="password"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Retype-Password"
              name="retypePassword"
              type="password"
            />
          </div>
        </div>
        <div
          className="row"
        />
        <div
          className="center-align"
        />
        <div
          className="row center-align"
        >
          <button
            className="btn btn-large waves-effect waves-light
                  indigo accent-2"
            disabled={false}
            name="signupbtn"
            type="submit"
          >
            Sign Up
          </button>
        </div>
        <div
          className="row center-align"
        >
          Already have an account? 
          <a
            className="cursor-pointer blue-text"
            onClick={[MockFunction]}
            role="button"
            tabIndex="0"
          >
            Sign In
          </a>
        </div>
      </form>
    </div>
  </Dialog>
</div>
`;

exports[`Signup component should show error in signup 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[MockFunction]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={true}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    modal={true}
    open={true}
    repositionOnUpdate={true}
    title="Create an account"
  >
    <div
      className="row"
    >
      <form
        className="col s12"
      >
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="First Name"
              name="firstName"
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Last Name"
              name="lastName"
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Username"
              name="username"
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Email"
              name="email"
              type="email"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Password"
              name="password"
              type="password"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Retype-Password"
              name="retypePassword"
              type="password"
            />
          </div>
        </div>
        <ErrorMessage
          message="Error in signup"
        />
        <div
          className="row"
        />
        <div
          className="center-align"
        />
        <div
          className="row center-align"
        >
          <button
            className="btn btn-large waves-effect waves-light
                  indigo accent-2"
            disabled={false}
            name="signupbtn"
            type="submit"
          >
            Sign Up
          </button>
        </div>
        <div
          className="row center-align"
        >
          Already have an account? 
          <a
            className="cursor-pointer blue-text"
            onClick={[MockFunction]}
            role="button"
            tabIndex="0"
          >
            Sign In
          </a>
        </div>
      </form>
    </div>
  </Dialog>
</div>
`;

exports[`Signup component should show spinner when submitting details 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[MockFunction]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={true}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    modal={true}
    open={true}
    repositionOnUpdate={true}
    title="Create an account"
  >
    <div
      className="row"
    >
      <form
        className="col s12"
      >
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="First Name"
              name="firstName"
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Last Name"
              name="lastName"
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Username"
              name="username"
              type="text"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Email"
              name="email"
              type="email"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Password"
              name="password"
              type="password"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <Field
              component={[Function]}
              label="Retype-Password"
              name="retypePassword"
              type="password"
            />
          </div>
        </div>
        <div
          className="row"
        />
        <div
          className="center-align"
        >
          <Spinner
            size="small"
          />
        </div>
        <div
          className="row center-align"
        >
          <button
            className="btn btn-large waves-effect waves-light
                  indigo accent-2"
            disabled={true}
            name="signupbtn"
            type="submit"
          >
            Sign Up
          </button>
        </div>
        <div
          className="row center-align"
        >
          Already have an account? 
          <a
            className="cursor-pointer blue-text"
            onClick={[MockFunction]}
            role="button"
            tabIndex="0"
          >
            Sign In
          </a>
        </div>
      </form>
    </div>
  </Dialog>
</div>
`;
